File: 35-search-insert-position/search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& arr, int target) {
        int l=0,h=arr.size()-1,mid,ans=arr.size();
        while(l<=h){
           mid=(l+h)/2;
           if(arr[mid]==target){
                return mid;
           }
           else if(arr[mid]<target){
                l=mid+1;
           }
           else
           h=mid-1;
        }
        return l;
    }
};

File: 35-search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] arr, int t) {
        int s=0,h=arr.length-1;
        while(s<=h){
            int mid=s+(h-s)/2;
            if(arr[mid]==t)
            return mid;
            else if(arr[mid]>t)
            h=mid-1;
            else
            s=mid+1;
        }
        return s;
    }
}